arch/Kconfig

```diff
@@ -414,6 +414,12 @@ config CPU_HAS_TEE
 	  Execution Environment (e.g. when it has a security attribution
 	  unit).
 
+config CPU_HAS_DCLS
```

Comment:
@stephanosio one more thing; i wonder if this is as-generic-as-possible.
AFAIK there can be multi-level redundant lockstep configuration, e..g triple modular, with error correction with majority vote. So I wonder if we could use a generic Kconfig name for lockstep here.

Reply:
@ioannisg Initially, I thought of adding something like `CPU_HAS_LOCKSTEP`, but decided to go with `CPU_HAS_DCLS` for now because "TCLS" or higher is simply unheard of and its practicality is questionable for general safety critical applications (i.e. automotive and industrial).

It is worth noting that, at the time of writing, there is not even a single commercially available solution that implements triple redundancy at core level (except the space-grade Cortex-R5 TCLS radiation-tolerant hack which is still WIP and only uses TCLS for RTL-level radiation tolerance instead of full fledged radiation hardening in PD), and it is almost always preferable to have multiple physically separate DCLS systems working in fail-over or majority voting configuration, as physical system component failure is a dominant factor when it comes to extreme reliability.

All that aside, DCLS and TCLS (or MCLS) handling can be quite different and I believe they should be specified using different symbols. For instance, the ARM DCLS requires software-level fault handling and core re-synchronisation for error recovery, whereas most of the error recovery process is handled in the hardware block for the TCLS (the space-grade implementation).

Reply:
Thanks for the explanation here @stephanosio 
So I am wondering; should we have this, for now, as an ARM-aRCH-only option?

Reply:
> So I am wondering; should we have this, for now, as an ARM-aRCH-only option?

@ioannisg As far as I can see, there are quite a few archs that are currently supported by the Zephyr that can potentially make use of this symbol:

1. ARC: https://www.synopsys.com/dw/ipdir.php?ds=arc-emfs
2. NIOS II: https://nmi.org.uk/wp-content/uploads/2016/05/Altera_NMI-Func-Safety.pdf
3. RISC-V (WIP): https://riscv.org/wp-content/uploads/2017/05/Wed0900am-RISC-V-Lockstep-Processor-Odiga.pdf